### PR TITLE
Created new pure-Java grex-core subproject, closes #3

### LINKED ISSUE #3: Drop the dependency on `Context`?
This is more of a tentative suggestion than an issue, but I wanted to get some opinion on it before I do any work on it and submit a pull request.

It seems slightly a shame that this library isn't "pure" Java. It would precisely fit a use case I have which isn't on Android, and the only thing preventing that is a single line of code!

I think it would be straightforward to create a `grex-core` library which the main `grex` library would then have as a dependency. And then the main `grex` would continue to be usable directly and transparently on Android, but people like me could use `grex-core` in a Java project and just provide the (hopefully tiny) implementation detail that would be needed.

The main problem with this approach is probably that it increases the maintenance load a non-trivial amount, but hopefully not too much. Apart from that I can't see any major problems with the idea, but just wanted to get some feedback before starting on it.

I'd also like to say for the record, I really like this library, and I'm hoping many more pop up providing an RxJava interface to useful functionality.

### PR DESCRIPTION
**Pull request by grodin**
I'd like a review on this please. The functionality should certainly be there, and I've got reasonable-ish test coverage.

Things not updated in this PR:

- README
- CHANGELOG
- No new Maven artifact, which there probably should be, not sure how that should go

I think I've matched your code style, but I can't be certain.

I've got a few more things I'd like to submit, but they're not really related to this, so I'll clean them up and send them as separate PRs. I could add them as issues as well if you'd like.

### CODE REVIEW

File: grex-core/src/main/java/au/com/gridstone/grex/BaseGRexPersister.java [hunk 1]
```diff
@@ -0,0 +1,380 @@
+package au.com.gridstone.grex;/*
```
Comment (chris-horner) [inline grex-core/src/main/java/au/com/gridstone/grex/BaseGRexPersister.java]: Copyright block comment should go above package definition.

File: grex-core/src/main/java/au/com/gridstone/grex/FileFactory.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
+/*
```
Comment (chris-horner) [inline grex-core/src/main/java/au/com/gridstone/grex/FileFactory.java]: This file doesn't appear to be used at all. Should probably be removed.

File: grex-core/src/main/java/au/com/gridstone/grex/FileIODelegate.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+            return new FileReader(file);
+        }
+    }
+
+    @Override
+    public Writer getWriter(final String key) throws IOException {
+        return new FileWriter(getFile(key));
+    }
+
+    @Override public boolean clear(final String key) {
```
Comment (chris-horner) [inline grex-core/src/main/java/au/com/gridstone/grex/FileIODelegate.java]: `@Override` should be on a line above the method declaration with the current style.

File: grex-core/src/main/java/au/com/gridstone/grex/FileFactory.java (hunk 2 above)
Comment (grodin) [inline grex-core/src/main/java/au/com/gridstone/grex/FileFactory.java]: Yep, oversight on my part.

**Comment by chris-horner** [comment]
Nice work with this!

What're your thoughts on turning `grex-core` into `grex`, and `grex` into `grex-android`. I know this library started out as being Android only, but that naming structure makes a little more sense to me.

**Comment by grodin** [comment]
The only reason I didn't do that in the first place was to avoid breaking the API again. It's the same reason that the Android `Persister` instance is called `GrexPersister` when ideally, I think that name should be used for the interface.

I'm generally in favour of not breaking APIs for published stuff if possible, but I certainly think the architecture makes more sense that way.

I'll fix the issues you noted above, and I'll do a version of this PR with the alternative architecture so we've got something concrete to compare with.

**Comment by grodin** [comment]
Style issues fixed.

**Comment by grodin** [comment]
Submitted PR #10 with the alternate naming.

**Pull request merged**